apps/oauth2/lib/Controller/OauthApiController.php

```diff
@@ -226,4 +230,32 @@ public function getToken(
 			]
 		);
 	}
+
+	/**
+	 * @PublicPage
+	 * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+	 */
+	public function getUserInfo() {
+		$user = $this->userSession->getUser();
+		if ($user) {
```

Comment:
```suggestion
		if ($user === null) {
		    return new JSONResponse([
				'error' => 'user_not_found',
			], Http::STATUS_NOT_FOUND);
```

make this an early return